### DISCUSSION TITLE
Phone fails to connect to mitmproxy running on one PC but connects to mitmproxy running on other machines

### DISCUSSION THREAD

**Discussion by charlbury** [mitmproxy/mitmproxy#6966] [Q&A]
Problem Summary: iPhone fails to connect to mitmproxy on a Windows 10 PC when other iphones succeed. It works connecting to another Windows 10 PC mitmproxy or mitmproxy on a Mac.

* The phone never gets a response browsing to mitm.it or anything else when the phone's wifi proxy is set to the problem windows machine running mitmproxy. I never see any traffic in mitmproxy.
* If I set the same phone's proxy to another windows machine running mitmproxy on the same wifi network, it does get a response from mitm.it and I see the traffic in mitmproxy. The only thing changed on the phone is the last three digits of the proxy server ip address.
* If I set the same phone's proxy to a mac running mitmproxy on the same wifi network, it does get a response from mitm.it and I see the traffic in mitmproxy.
* If I connect another iphone to the problem windows machine proxy it does get a response when browsing to mitm.it and I see its traffic in mitmproxy.

Question: I haven't messed around with the network or firewall settings on the phone or the Windows pc but are there any additional settings on the iphone or the windows machine (like firewall settings etc.) that I should check to see why one device does not get a connection but another one does?

Hardware Details:
* mitmproxy running on
  * Windows 10 PCs
  * Macbook pro 2019 intel MacOS Sonoma 14 
* Phones
  * Problem phone: iPhone XR, iOS 16.7
  * Working phone: iPhone 13, iOS 17.5.1

**Comment by charlbury** [comment]
Well, I ended up resetting the PC and reinstalling windows then resetting the iphone to factory defaults. Installing mitmproxy and setting the phones proxy to point it then worked. Sure there must have been a network setting that could have been reset to avoid this, but couldn't find anything that made a difference.

**Comment by charlbury** [comment]
So now what happens on this clean WIndows and iPhone install is, mitmproxy will stop intercepting the requests after X number of requests or Y amount of time. I don't know if it's number of requests or time, but, it stops intercepting the requests as the phone no longer has a connection to mitmproxy.

mitmproxy binary 10.3.1

**Reply by rosydawn6** [reply]
You should turn on verbose logging on mitmproxy to determine if there is any pattern or indication of what is happening

**Reply by charlbury** [reply]
Thanks for suggestion, will do this.
